lib/xapi-stdext-unix/unixext.ml

```diff
@@ -409,26 +409,52 @@ let string_of_signal x =
   then List.assoc x table
   else (Printf.sprintf "(ocaml signal %d with an unknown name)" x)
 
+let with_polly f =
+  let polly = Polly.create () in
+  let finally () = Polly.close polly in
+  Xapi_stdext_pervasives.Pervasiveext.finally
+  (fun () -> f polly) finally
+
+
 let proxy (a: Unix.file_descr) (b: Unix.file_descr) =
   let size = 64 * 1024 in
   (* [a'] is read from [a] and will be written to [b] *)
   (* [b'] is read from [b] and will be written to [a] *)
   let a' = CBuf.empty size and b' = CBuf.empty size in
   Unix.set_nonblock a;
   Unix.set_nonblock b;
+  with_polly @@ fun polly ->
 
   try
     while true do
```

Comment:
could this be a recursive function with `any` as a parameter which is used for termination?